packages/autocomplete-plugin-recent-searches/src/createRecentSearchesPlugin.ts

```diff
@@ -0,0 +1,86 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-core';
+
+import { createRecentSearchesStore } from './createRecentSearchesStore';
+
+type PluginOptions = {
+  /**
+   * The number of searches to store.
+   *
+   * @default 5
+   */
+  limit?: number;
+
+  /**
+   * The key to distinguish multiple stores of recent searches.
+   *
+   * @example
+   * // 'top_searchbar'
+   */
+  key: string;
+};
+
+type RecentSearchesPlugin<TItem> = AutocompletePlugin<TItem> & {
+  data: {
+    getFacetFilters: () => string[];
+  };
+};
+
+export function createRecentSearchesPlugin<TItem>({
+  key,
+  limit = 5,
+}: PluginOptions): RecentSearchesPlugin<TItem> {
+  const store = createRecentSearchesStore({
+    key: ['AUTOCOMPLETE_RECENT_SEARCHES', key].filter(Boolean).join('__'),
```

Comment:
Since the key is required now, no need to filter.

```suggestion
    key: ['AUTOCOMPLETE_RECENT_SEARCHES', key].join('__'),
```